docs/sources/clients/promtail/configuration.md

```diff
@@ -1622,6 +1647,99 @@ users with thousands of services it can be more efficient to use the Consul API
 directly which has basic support for filtering nodes (currently by node
 metadata and a single tag).
 
+### docker_sd_config
+
+Docker service discovery configurations allow retrieving targets from a Docker daemon.
+It will only watch containers of the daemon referenced with the host parameter. That
+means it should run on each node in a distributed setup. The containers must run with
+either the [json-file](https://docs.docker.com/config/containers/logging/json-file/)
+or [journald](https://docs.docker.com/config/containers/logging/journald/) logging driver.
+
+Please note that the discovery will not pick up finished containers. That means
+Promtail will not scrape the remaining logs from finished containers after a restart.
+
+The configuration is inherited from [Prometheus' Docker service discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config).
+
+```yaml
+# Address of the Docker daemon.  Use unix:///var/run/docker.sock for a local setup.
+host: <string>
+
+# Optional proxy URL.
+[ proxy_url: <string> ]
+
+# TLS configuration.
+tls_config:
+  [ <tls_config> ]
+
+# The port to scrape metrics from, when `role` is nodes, and for discovered
+# tasks and services that don't have published ports.
+[ port: <int> | default = 80 ]
+
+# The host to use if the container is in host networking mode.
+[ host_networking_host: <string> | default = "localhost" ]
+
+# Optional filters to limit the discovery process to a subset of available
+# resources.
+# The available filters are listed in the upstream documentation:
+# Containers: https://docs.docker.com/engine/api/v1.41/#operation/ContainerList
+[ filters:
+  [ - name: <string>
+      values: <string>, [...] ]
+]
+
+# The time after which the containers are refreshed.
+[ refresh_interval: <duration> | default = 60s ]
+
+# Authentication information used to authenticate to the Docker daemon.
+# Note that `basic_auth` and `authorization` options are
+# mutually exclusive.
+# password and password_file are mutually exclusive.
+
+# Optional HTTP basic authentication information.
+basic_auth:
+  [ username: <string> ]
+  [ password: <secret> ]
+  [ password_file: <string> ]
+
+# Optional `Authorization` header configuration.
+authorization:
+  # Sets the authentication type.
+  [ type: <string> | default: Bearer ]
+  # Sets the credentials. It is mutually exclusive with
+  # `credentials_file`.
+  [ credentials: <secret> ]
+  # Sets the credentials to the credentials read from the configured file.
+  # It is mutually exclusive with `credentials`.
+  [ credentials_file: <filename> ]
+
+# Optional OAuth 2.0 configuration.
+# Cannot be used at the same time as basic_auth or authorization.
+oauth2:
+  [ <oauth2> ]
+
+# Configure whether HTTP requests follow HTTP 3xx redirects.
+[ follow_redirects: <bool> | default = true ]
+```
+
+Available meta labels:
+
+  * `__meta_docker_container_id`: the id of the container
```

Comment:
Would be great to have a full good example, with correct relabeling so people can copy/paste.